### DISCUSSION TITLE
How to force the use of security authentication when registering a wazuh agent?

### DISCUSSION THREAD

**Discussion by liurx1994** [wazuh/wazuh#15629] [Q&A]
The wazuh server is configured with mTLS authentication and password authentication, but the agent can still register successfully without configuring a certificate or password when registering.

**Comment by jctello** [comment]
Hi @liurx1994 ,
This is likely related to the issue reported [here](https://github.com/wazuh/wazuh/issues/13772) which overlooked was promptly resolved on version 4.3.5. 
I have verified the verification of the agent's certificate by following [this documentation](https://documentation.wazuh.com/current/user-manual/agent-enrollment/security-options/agent-identity-verification.html) on Wazuh version 4.3.10 and verified that if no certificate is provided the agent's `ossec.log` will indicate:
```
2022/12/07 09:07:11 wazuh-agentd: INFO: Requesting a key from server: mywazuhmanager.com
2022/12/07 09:07:11 wazuh-agentd: INFO: No authentication password provided
2022/12/07 09:07:11 wazuh-agentd: INFO: Using agent name as: nocertagent
2022/12/07 09:07:11 wazuh-agentd: INFO: Waiting for server reply
2022/12/07 09:07:11 wazuh-agentd: ERROR: SSL read (unable to receive message)
2022/12/07 09:07:11 wazuh-agentd: ERROR: If Agent verification is enabled, agent key and certificates may be incorrect!
```
And if a certificate for the wrong IP is provided the agent's `ossec.log` will state:
```
2022/12/07 09:21:18 wazuh-agentd: INFO: Requesting a key from server: mywazuhmanager.com
2022/12/07 09:21:18 wazuh-agentd: INFO: No authentication password provided
2022/12/07 09:21:18 wazuh-agentd: INFO: Using agent name as: smith
2022/12/07 09:21:18 wazuh-agentd: INFO: Waiting for server reply
2022/12/07 09:21:18 wazuh-agentd: ERROR: SSL read (unable to receive message)
2022/12/07 09:21:18 wazuh-agentd: ERROR: If Agent verification is enabled, agent key and certificates may be incorrect!
```
Whilst the manager's `ossec.log` reflects:
```
2022/12/07 09:17:57 wazuh-authd: INFO: New connection from myagentrouter.com
2022/12/07 09:17:57 wazuh-authd: ERROR: Unable to verify client certificate.
```

With the correct certificate the agent registers as expected. 

Note that I'm using both:
```
    <ssl_agent_ca>/var/ossec/etc/rootCA.pem</ssl_agent_ca>
    <ssl_verify_host>yes</ssl_verify_host>
```
on the manager's `<auth>` section of the `ossec.conf`

So in order to force the security authentication of certificate registration please upgrade to the most recent version of Wazuh.
